python/lsst/ip/isr/isrTask.py

```diff
@@ -696,44 +1151,222 @@ def convertIntToFloat(self, exposure):
+        if numpy.all(maskView.getArray() & maskVal > 0):
+            badAmp = True
+
+        return badAmp
+
+    def overscanCorrection(self, ccdExposure, amp):
+        """Apply overscan correction in place.
+
+        This method does initial pixel rejection of the overscan
+        region.  The overscan can also be optionally segmented to
+        allow for discontinuous overscan responses to be fit
+        separately.  The actual overscan subtraction is performed by
+        the `lsst.ip.isr.isrFunctions.overscanCorrection` function,
+        which is called here after the amplifier is preprocessed.
 
-        Checks config.doLinearize and the linearity type of the first amplifier.
+        Parameters
+        ----------
+        ccdExposure : `lsst.afw.image.Exposure`
+            Exposure to have overscan correction performed.
+        amp : `lsst.afw.table.AmpInfoCatalog`
+            The amplifier to consider while correcting the overscan.
 
-        @param[in]  detector  detector information (an lsst.afw.cameraGeom.Detector)
+        Returns
+        -------
+        overscanResults : `lsst.pipe.base.Struct`
+            Result struct with components:
+            - ``imageFit`` : scalar or `lsst.afw.image.Image`
+                Value or fit subtracted from the amplifier image data.
+            - ``overscanFit`` : scalar or `lsst.afw.image.Image`
+                Value or fit subtracted from the overscan image data.
+            - ``overscanImage`` : `lsst.afw.image.Image`
+                Image of the overscan region with the overscan
+                correction applied. This quantity is used to estimate
+                the amplifier read noise empirically.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if the ``amp`` does not contain raw pixel information.
+
+        See Also
+        --------
+        lsst.ip.isr.isrFunctions.overscanCorrection
         """
-        return self.config.doLinearize and \
-            detector.getAmpInfoCatalog()[0].getLinearityType() != NullLinearityType
+        if not amp.getHasRawInfo():
+            raise RuntimeError("This method must be executed on an amp with raw information.")
+
+        if amp.getRawHorizontalOverscanBBox().isEmpty():
+            self.log.info("ISR_OSCAN: No overscan region.  Not performing overscan correction.")
+            return None
+
+        # Construct views
+        ampImage = afwImage.MaskedImageF(ccdExposure.getMaskedImage(), amp.getRawDataBBox(),
+                                         afwImage.PARENT)
+        overscanImage = afwImage.MaskedImageF(ccdExposure.getMaskedImage(),
+                                              amp.getRawHorizontalOverscanBBox(),
+                                              afwImage.PARENT)
+        overscanArray = overscanImage.getImage().getArray()
+
+        statControl = afwMath.StatisticsControl()
+        statControl.setAndMask(ccdExposure.getMaskedImage().getMask().getPlaneBitMask("SAT"))
```

Comment:
Is this just a rendering error?  It seems like this line is indented far too far.

Reply:
Oh never mind.  My browser window was too narrow.